src/coreclr/jit/compiler.hpp

```diff
@@ -1466,6 +1466,7 @@ inline GenTreeLclVar* GenTree::BashToLclVar(Compiler* comp, unsigned lclNum)
     ChangeOper(GT_LCL_VAR);
     ChangeType(varDsc->lvNormalizeOnLoad() ? varDsc->TypeGet() : genActualType(varDsc));
     AsLclVar()->SetLclNum(lclNum);
+    INDEBUG(AsLclVar()->ResetLclILoffs());
```

Comment:
`ResetLclILoffs` should be done as part of `SetOper` instead of being scattered around the callers.

Reply:
I thought to do in `ChangeOper()` but that would mean I would have to check for `oper == GT_LCL_VAR`, but I suppose since this is `DEBUG`, it shouldn't matter much and may be `SetOper()` is more natural place. I will move it.